File: java/58_LengthofLastWord.java
class Solution {
    public int lengthOfLastWord(String s) {
        int len = s.length();
        int ans = 0;
        for(int i = len - 1; i >= 0; i--) {
        	if(s.charAt(i) == ' ') {
        		len--;
        	}else {
        		break;
        	}
        }
        for(int i = len - 1; i >= 0; i--) {
        	if(s.charAt(i) != ' ') {
        		ans++;
        	}else {
				break;
			}
        }
        return ans;
    }
}

File: c++/58_LengthofLastWord.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.size();
        int k = n - 1;
        while (k >= 0 && s[k] == ' ') k--;
        int j = k;
        while (j >= 0 && s[j] != ' ') j--;
        return k - j;
    }
};